cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -661,15 +661,43 @@ func (m *gceManagerImpl) fetchExplicitMigs(specs []string) error {
 	return nil
 }
 
-func (m *gceManagerImpl) buildMigFromSpec(spec string) (*Mig, error) {
-	s, err := dynamic.SpecFromString(spec, scaleToZeroSupported)
+func (m *gceManagerImpl) buildMigFromFlag(flag string) (Mig, error) {
+	s, err := dynamic.SpecFromString(flag, scaleToZeroSupported)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse node group spec: %v", err)
 	}
-	mig := &Mig{gceManager: m, minSize: s.MinSize, maxSize: s.MaxSize, exist: true}
-	if mig.Project, mig.Zone, mig.Name, err = ParseMigUrl(s.Name); err != nil {
+	return m.buildMigFromSpec(s)
+}
+
+func (m *gceManagerImpl) buildMigFromAutoCfg(link string, cfg cloudprovider.MIGAutoDiscoveryConfig) (Mig, error) {
+	s := &dynamic.NodeGroupSpec{
+		Name:               link,
+		MinSize:            cfg.MinSize,
+		MaxSize:            cfg.MaxSize,
+		SupportScaleToZero: scaleToZeroSupported,
+	}
+	return m.buildMigFromSpec(s)
+}
+
+func (m *gceManagerImpl) buildMigFromSpec(s *dynamic.NodeGroupSpec) (Mig, error) {
+	if err := s.Validate(); err != nil {
```

Comment:
This only happened in buildMigFromAutoCfg before, but not for explicitly configured migs. It sounds like a reasonable thing to do, but I have no idea what it really does. Are you sure this is ok?

Reply:
Validate() is a basic sanity check: node group name must not be empty, maxSize >= minSize, minSize > 0 (or >= 0 if scale to zero isn't supported.)  It's true we never checked any of this for explicitly configured migs before, which is all the more reason I'd like to add it.

https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/config/dynamic/node_group_spec.go#L68

Reply:
It seems we actually do check it already: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/config/dynamic/node_group_spec.go#L60. We call SpecFromString immediately before buildMigFromSpec, so with your change we end up calling Validate() twice for explicitly configured MIGs. I guess it doesn't matter all that much in that case.